### DISCUSSION TITLE
Dienstbeschreibung

### DISCUSSION THREAD

**Discussion by marko-jer** [e-netze/webgis-community#88] [General]
Wir konfigurieren unsere Dienste im ArcGIS Pro. Da kann man die Beschreibung des Dienstes in Map-Properties/Metadata hinzufügen, die vom Dienst als Service Description übernommen wird. Jedoch wird diese, wie es aussieht mit html-tags formatiert.

Für die **Beschreibung in der Karte** passt es:

![image](https://github.com/gis-eni/webgis-forum/assets/1842500/43ea39dc-6aba-4714-8984-6c5599743e98)

Im Kartenviewer  unter "Info & Copyright" bzw. beim "Dienste hinzufügen" **Dienstbeschreibung** werden die html-tags nicht berücksichtigt:

![image](https://github.com/gis-eni/webgis-forum/assets/1842500/e9f1d4ab-4aed-42a4-809d-9c584b09dfcc)

Alternative für uns mit ArcGIS Pro wäre sonst nur, nach dem Publizieren im AGS-Manager die Beschreibung eingeben- ist umständlich.

**Comment by gis-eni** [comment]
Das ist absichtlich so gemacht:
bei den Injection Tests und Lösungen wurden Dinge wie das direkte einfügen von HTML Code in den Viewer aus unsicheren Quellen unterbunden. Damit könnte potenziell Schadcode auf die Seite kommen. Es ist nicht garantiert, dass die Dienste im Viewer immer aus deinen AGS kommen. Kommt der Dienst von dritten, könnte man so Javascript auf die Seite einschleusen etc.

Warum AGS genau diesen Weg einschlägt ist mir schleierhaft :/

Interessant wäre einmal so ein Dienst zum testen. Kannst du (per Mail) so etwas bereit stellen?
Dann könnte man theoretisch probieren, das HTML schon am Server in etwas harmloseres zu konvertieren (zB Markdown). Das hier auch noch inline-styles drinnen stehen ist generell auch schlecht. Um die Formatierung sollte sich eigentlich Seite kümmern, in der die Description eingefügt wird.
Macht ArcGIS Pro das automatisch (mit den Styles) oder machst du das im ArcGIS Pro (Schriftarten, fett, kursiv, etc)

**Comment by marko-jer** [comment]
Ich verstehe die Sicherheitsbedenken!
Warum über ArcGIS Pro eine Formatierung für die Beschreibung der Dienste möglich ist ... kann ich auch nicht nachvollziehen.
Hier noch zwei screenshots 

1.  Eingabe im ArcGIS Pro:
![image](https://github.com/gis-eni/webgis-forum/assets/1842500/bf758ed1-49c9-4987-ada9-76c398359073)

2. so beim Publizieren - da sieht man die Beschreibung gar nicht mehr:
![image](https://github.com/gis-eni/webgis-forum/assets/1842500/3a0e6ac5-a055-4137-805a-d25ab324f07f)

3. und so sieht es dann im AGS-Manager aus:
 ![image](https://github.com/gis-eni/webgis-forum/assets/1842500/56386b5e-fd69-4994-93d4-fcfb84e30948)

Man kann im ArcGIS Pro keine zusätzlichen html-Tags einfügen. Ich nehme an, so wie du schreibst, wird hier noch was harmloseres verwendet, das beim publizieren in html-Tags umgesetzt wird - siehe 3. Bild.

Ein unpraktischer Workaround: keine Beschreibung im ArcGIS Pro eingeben und die Beschreibung erst nach dem Publizieren im AGS-Manager eingeben  ;(

**Comment by gis-eni** [comment]
Ok, die Description so zu parsen, dass alle möglichen Stile berücksichtigt werden ist leider nicht einfach möglich. Ich kann einen HTML Rendering Engine nachbauen, die alles berücksichtig und in einen anderes Format (Markdown) umwandelt.

Fett/Kursiv wird nicht hinhauen.

Links sollten funktionieren. Wenn ich deinen Dienst nehme sieht das bei dann so aus:

![image](https://github.com/gis-eni/webgis-forum/assets/62870568/8c358a3c-c860-4386-841e-51eff0ac030d)


Eigentlich bis auf fett/kursiv wie in deiner Vorlage.
Der Link funktioniert und die Seite wird in einem neuen Tab geöffnet.

**Reply by marko-jer** [reply]
Das passt - fett/kursiv/unterstrichen ist nicht wichtig - Links dagegen schon. DANKE!
Ein bisschen auseinandergezogen schaut es noch aus - kannst du  `<p></p>` als `</br>` übersetzen, dann schaut es kompakter aus?
